src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -195,4 +193,20 @@ private void storeMyUserId(final Object channelData) {
         final Map channelDataMap = new Gson().fromJson((String)channelData, Map.class);
         myUserID = String.valueOf(channelDataMap.get("user_id"));
     }
+
+    private class MemberData {
+        public String user_id;
+        public Object user_info;
```

Comment:
`Gson` object is thread safe (https://groups.google.com/forum/#!topic/google-gson/Vju1HuJJUIE) so create one with the correct policy and assign to a constant, use it throughout the class.